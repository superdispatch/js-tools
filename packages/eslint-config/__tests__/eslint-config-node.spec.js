const { testInheritance } = require('../__testutils__/test-eslint-config');

const configFile = require.resolve('../node');
const baseConfigFile = require.resolve('../base');

testInheritance(configFile, baseConfigFile);
