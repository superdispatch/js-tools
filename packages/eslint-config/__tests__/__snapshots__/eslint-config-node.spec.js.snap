// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inheritance config 1`] = `
Snapshot Diff:
- First value
+ Second value

  Object {
-   "env": Object {},
+   "env": Object {
+     "node": true,
+   },
    "globals": Object {},
    "parser": null,
    "parserOptions": Object {
      "ecmaVersion": 2018,
    },
    "plugins": Array [
      "prettier",
      "import",
+     "node",
    ],
    "settings": Object {},
  }
`;

exports[`Inheritance dev rules 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`Inheritance rules 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
    ],
+   "import/no-extraneous-dependencies": Array [
+     2,
+     Object {
+       "devDependencies": true,
+       "optionalDependencies": false,
+       "peerDependencies": false,
+     },
+   ],
    "import/no-mutable-exports": Array [
@@ --- --- @@
    "no-console": Array [
-     2,
+     0,
    ],
@@ --- --- @@
      "off",
+   ],
+   "node/exports-style": Array [
+     2,
+   ],
+   "node/no-deprecated-api": Array [
+     2,
+   ],
+   "node/no-extraneous-import": Array [
+     2,
+   ],
+   "node/no-extraneous-require": Array [
+     2,
+   ],
+   "node/no-missing-import": Array [
+     2,
+   ],
+   "node/no-missing-require": Array [
+     2,
+   ],
+   "node/no-unpublished-bin": Array [
+     2,
+   ],
+   "node/no-unpublished-import": Array [
+     2,
+   ],
+   "node/no-unpublished-require": Array [
+     2,
+   ],
+   "node/no-unsupported-features/es-builtins": Array [
+     2,
+   ],
+   "node/no-unsupported-features/es-syntax": Array [
+     2,
+   ],
+   "node/no-unsupported-features/node-builtins": Array [
+     2,
+   ],
+   "node/prefer-global/buffer": Array [
+     2,
+   ],
+   "node/prefer-global/console": Array [
+     2,
+   ],
+   "node/prefer-global/process": Array [
+     2,
+   ],
+   "node/prefer-global/text-decoder": Array [
+     2,
+   ],
+   "node/prefer-global/text-encoder": Array [
+     2,
+   ],
+   "node/prefer-global/url": Array [
+     2,
+   ],
+   "node/prefer-global/url-search-params": Array [
+     2,
+   ],
+   "node/prefer-promises/dns": Array [
+     2,
+   ],
+   "node/prefer-promises/fs": Array [
+     2,
+   ],
+   "node/process-exit-as-throw": Array [
+     2,
+   ],
+   "node/shebang": Array [
+     2,
    ],
`;
